R/z_pages.R

```diff
@@ -35,12 +36,17 @@
 #' }
 #'
 #' @export
-animint2pages <- function(plot.list, github_repo, commit_message = "Commit from animint2pages", private = FALSE, required_opts = c("title","source"), ...) {
+
+
+
+animint2pages <- function(plot.list, github_repo, commit_message = "Commit from animint2pages", private = FALSE, required_opts = c("title","source"),server=NULL, ...) {
```

Comment:
Hi @siddhesh195 , can you provide more details on how the param `server` works?

Reply:
@Faye-yufan server parameter is an object of persistent server which should be created in every test file having animint2pages and it remains running when started. This avoids the need to create a new server in every call of animint2pages. But if you see gh-pages test where I initialize server in the beginning, you will notice that I create a new R session for the server. This is because directly using servr API did not work properly for some reason to handle dynamic html even though it says it is in non blocking daemon mode. But having a separate R session makes it complex to terminate the server, so it just keeps running until that R environment is closed.

Reply:
> @Faye-yufan server parameter is an object of persistent server which should be created in every test file having animint2pages and it remains running when started. This avoids the need to create a new server in every call of animint2pages. But if you see gh-pages test where I initialize server in the beginning, you will notice that I create a new R session for the server. This is because directly using servr API did not work properly for some reason to handle dynamic html even though it says it is in non blocking daemon mode. But having a separate R session makes it complex to terminate the server, so it just keeps running until that R environment is closed.

if the server object is passed in aminint2pages, we need to tell animint2dir to use the same directory that the server is already pointing to, for that to work I had to make some changes to animint2dir function too.

Reply:
please add space after comma: `, server=NULL`

also please add docs for new server argument